app/models/concerns/reviewable_action_builder.rb

```diff
@@ -3,6 +3,97 @@
 module ReviewableActionBuilder
   extend ActiveSupport::Concern
 
+  # Indicates whether the reviewable can be flagged. Flaggable reviewable types
+  # should override this method and return true.
+  #
+  # @return [Boolean]
+  def flaggable?
+    false
```

Comment:
What do you think about using that register `Reviewable.flaggable_types` instead of methods?

So we would need to add `ReviewablePost` and `ReviewableFlaggedPost` to register but we would have one place defining which types are flaggable.